sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/trees/TreeNode.scala

```diff
@@ -807,7 +807,7 @@ abstract class TreeNode[BaseType <: TreeNode[BaseType]] extends Product with Tre
       truncatedString(seq.map(formatArg(_, maxFields)), "[", ", ", "]", maxFields) :: Nil
     case set: Set[_] =>
       // Sort elements for deterministic behaviours
-      val sortedSeq = set.toSeq.map(formatArg(_, maxFields).sorted)
+      val sortedSeq = set.toSeq.map(formatArg(_, maxFields)).sorted
```

Comment:
What about the same case in `formatArg` above?

Reply:
Yeah, the implementation has a lot of duplicated code...

Would you be OK with applying the fix to `formatArg` and then DRYing `argString` as follows.

**Before:**
```scala
    case seq: Seq[_] =>
      truncatedString(seq.map(formatArg(_, maxFields)), "[", ", ", "]", maxFields) :: Nil
    case set: Set[_] =>
      // Sort elements for deterministic behaviours
      val sortedSeq = set.toSeq.map(formatArg(_, maxFields)).sorted
      truncatedString(sortedSeq, "{", ", ", "}", maxFields) :: Nil
    case array: Array[_] if array.isEmpty => Nil
    case array: Array[_] =>
      truncatedString(array.map(formatArg(_, maxFields)), "[", ", ", "]", maxFields) :: Nil
```

**After:**
```scala
    case array: Array[_] if array.isEmpty => Nil
    case xs @ (_: Seq[_] | _: Set[_] | Array[_]) =>
      formatArg(_, maxFields) :: Nil
```

It's a bigger change than the minimum required, but it'll clean thing up a bit.

Reply:
I would prefer to just move `.sorted` (**Before**) instead of adding some more changes but I am fine with **After** versions if we're 100% sure input/output are same as expected.